### PR TITLE
feat: drop support for 32bit arch

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

Followup to https://github.com/elastic/apm-server/pull/11868

Packaging is currently broken because we're trying to build a 32bit artifact.

This PR removes support for 32 bit artifacts completely.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

File: packaging.mk [hunk 1]
```diff
@@ -106,25 +106,21 @@ WINDOWS_PACKAGE_FILES := \
...
 
-DEB_ARCH := i386 amd64 arm64
+DEB_ARCH := amd64 arm64
 DEBS := $(patsubst %, $(DISTDIR)/apm-server-$(APM_SERVER_VERSION)-%.deb, $(DEB_ARCH))
 DEBS += $(patsubst %, $(DISTDIR)/apm-server-$(APM_SERVER_VERSION)-SNAPSHOT-%.deb, $(DEB_ARCH))
-DEBS_386 := $(filter %-i386.deb, $(DEBS))
 DEBS_AMD64 := $(filter %-amd64.deb, $(DEBS))
 DEBS_ARM64 := $(filter %-arm64.deb, $(DEBS))
 
 RPM_ARCH := i686 x86_64 aarch64
```
Comment (axw) [inline packaging.mk:118]:
```suggestion
RPM_ARCH := x86_64 aarch64
```

File: packaging.mk [hunk 2]
```diff
@@ -182,12 +175,9 @@ $(DISTDIR)/%-docker-build-context.tar.gz: $(ARCHIVES_DIR)/%-docker-build-context
...
 	linux-x86_64.tar.gz \
-	linux-x86.tar.gz \
 	linux-arm64.tar.gz \
 	windows-x86_64.zip \
-	windows-x86.zip \
 	amd64.deb \
-	i386.deb \
 	arm64.deb \
 	x86_64.rpm \
 	i686.rpm \
```
Comment (axw) [inline packaging.mk:183]:
```suggestion
```

**Review by axw — APPROVED** [review]
LGTM, just a few more stragglers

File: x-pack/apm-server/aggregation.go [hunk 3]
```diff
@@ -2,8 +2,6 @@
 // or more contributor license agreements. Licensed under the Elastic License 2.0;
 // you may not use this file except in compliance with the Elastic License 2.0.
 
-//go:build !386
```
Comment (carsonip) [inline x-pack/apm-server/aggregation.go:5]: oops my bad

**Review by carsonip — APPROVED** [review]

**Pull request merged**